src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -512,30 +512,45 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
 
             foreach ((var resourceDataSchemaName, var operationSets) in ResourceDataSchemaNameToOperationSets)
             {
-                var resourceOperationsList = FindResourceToChildOperationsMap(operationSets);
-                foreach ((var operationSet, var operations) in resourceOperationsList)
+                foreach (var operationSet in operationSets)
                 {
-                    var isSingleton = operationSet.IsSingletonResource();
+                    var operations = GetChildOperations(operationSet.RequestPath);
                     // get the corresponding resource data
                     var originalResourcePath = operationSet.GetRequestPath();
                     var resourceData = GetResourceData(originalResourcePath);
-                    // we calculate the resource type of the resource
-                    var resourcePaths = originalResourcePath.Expand();
-                    foreach (var resourcePath in resourcePaths)
-                    {
-                        var resourceType = resourcePath.GetResourceType();
-                        var resource = new Resource(operationSet, operations, GetResourceName(resourceDataSchemaName, operationSet, resourcePath), resourceType, resourceData);
-                        var collection = isSingleton ? null : new ResourceCollection(operationSet, operations, resource);
-                        resource.ResourceCollection = collection;
-
-                        requestPathToResources.Add(resourcePath, new ResourceObjectAssociation(resourceType, resourceData, resource, collection));
-                    }
+                    if (operationSet.Count > 0)
```

Comment:
Why will there be an empty operation set?

Reply:
Yes, the operationSet of a `VirtualResource` is usually empty.